crates/pypi-types/src/lenient_requirement.rs

```diff
@@ -136,11 +158,28 @@ impl From<LenientRequirement> for Requirement {
 #[derive(Debug, Clone, Serialize, Eq, PartialEq)]
 pub struct LenientVersionSpecifiers(VersionSpecifiers);
 
+impl TrackedFromStr for LenientVersionSpecifiers {
+    type Err = VersionSpecifiersParseError;
+
+    fn tracked_from_str(
+        input: &str,
+        source: Option<&Path>,
+        working_dir: Option<&Path>,
+    ) -> Result<Self, Self::Err> {
+        Ok(Self(parse_with_fixups(
+            input,
+            "version specifier",
+            source,
+            working_dir,
+        )?))
+    }
+}
```

Comment:
Do you mind if I remove all these impls for `LenientVersionSpecifiers`, `LenientRequirement`, etc.? My read is that the path is always set to `None` as-is.

Reply:
(I did it but as a standalone commit, can revert if we decide against.)

Reply:
I added all of those in earlier on when I was just trying to make sure the source data didn't get lost at any point so added it everywhere. I think you're right, and if the tests pass with them removed, all good.